app/src/logging/config.py

```diff
@@ -26,41 +30,77 @@ class Config:
         env_nested_delimiter = "__"
 
 
-def configure_logging() -> logging.Handler:
-    """Configure logging for the application.
+class LoggingContext(ContextManager[None]):
+    """
+    A context manager for handling setting up the logging stream.
+
+    To help facillitate being able to test logging, we need to be able
+    to easily create temporary output streams and then tear them down.
+
+    When this context manager is torn down, the stream handler created
+    with it will be removed.
+
+    For example:
+    ```py
+    import logging
+
+    logger = logging.getLogger(__name__)
 
-    Configures the root module logger to log to stdout.
-    Adds a PII mask filter to the root logger.
-    Also configures log levels third party packages.
+    with LoggingContext("example_program_name"):
+        # This log message will go to stdout
+        logger.info("example log message")
+
+    # This log message won't go to stdout as the
+    # handler will have been removed
+    logger.info("example log message")
+    ```
+    Note that any other handlers added to the root logger won't be affected
+    and calling this multiple times before exit would result in duplicate logs.
     """
-    config = LoggingConfig()
-
-    # Loggers can be configured using config functions defined
-    # in logging.config or by directly making calls to the main API
-    # of the logging module (see https://docs.python.org/3/library/logging.config.html)
-    # We opt to use the main API using functions like `addHandler` which is
-    # non-destructive, i.e. it does not overwrite any existing handlers.
-    # In contrast, logging.config.dictConfig() would overwrite any existing loggers.
-    # This is important during testing, since fixtures like `caplog` add handlers that would
-    # get overwritten if we call logging.config.dictConfig() during the scope of the test.
-    console_handler = logging.StreamHandler(sys.stdout)
-    formatter = get_formatter(config)
-    console_handler.setFormatter(formatter)
-    console_handler.addFilter(pii.mask_pii)
-    logging.root.removeHandler(console_handler)
-    logging.root.addHandler(console_handler)
-    logging.root.setLevel(config.level)
-
-    if config.enable_audit:
-        src.logging.audit.init()
-
-    # Configure loggers for third party packages
-    logging.getLogger("alembic").setLevel(logging.INFO)
-    logging.getLogger("werkzeug").setLevel(logging.WARN)
-    logging.getLogger("sqlalchemy.pool").setLevel(logging.INFO)
-    logging.getLogger("sqlalchemy.dialects.postgresql").setLevel(logging.INFO)
-
-    return console_handler
+
+    def __init__(self, program_name: str) -> None:
+        self._configure_logging()
+        log_program_info(program_name)
+
+    def __enter__(self) -> None:
+        return None
+
+    def __exit__(self, exc_type: Any, exc_val: Any, exc_tb: Any) -> None:
+        # Remove the console handler to stop logs from being sent to stdout
```

Comment:
nit: is it worth explaining a little bit more the value of cleaning up? e.g. "This is useful in the test suite, since through the testing session the logger may be configured multiple times, so we want to clean up after each time" or something like that (feel free to make it less wordy)